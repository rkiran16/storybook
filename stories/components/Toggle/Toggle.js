import { html} from "lit-html";


export const Toggle = ({disabled = false, checked = false, variation, label = ""}) => {
   return html`
    <div class=${`form-check form-switch ${variation}`}>
        <input class="form-check-input" ?disabled=${disabled} ?checked=${checked} type="checkbox" id="flexSwitchCheckDefault" aria-label="Toggle Switch" aria-hidden="true">
        ${label !== "" ? html`<label class="form-check-label" for="flexSwitchCheckDefault">${label}</label>` : ""} 
    </div>
   `;
}