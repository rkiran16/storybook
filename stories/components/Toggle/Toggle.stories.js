import { Toggle } from "./Toggle";

export default {
  title: "Components/Toggle",
  parameters: {
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62e2e6811adf0a64d7db8a1c",
  },
  argTypes: {
    disabled: {
      name: "Disabled State",
      type: { name: 'boolean', required: false},
      description: "Use when the user cannot proceed or until an input is collected",
      defaultValue: false,
      table: {
        type: { summary: 'boolean' },
        defaultValue: { summary: false },
      },
      control: {
        type: 'boolean'
      }
    },
    checked: {
      name: "On State",
      type: { name: 'boolean', required: false},
      description: "when switch is turned on or auto switch",
      defaultValue: false,
      table: {
         type: { summary: 'boolean' },
         defaultValue: { summary: false },
      },
      control: { type: 'boolean'}
    },
    variation: {
      name: "Toggle Variations",
      defaultValue: "default",
      description: "Toggle Size Variations",
      control: { type: 'radio'},
      table: {
         type: { summary: 'radio' },
         defaultValue: { summary: 'default' },
      },
      options: ['default', 'large'],
      mapping: {
         default: 'toggle',
         large: 'toggle toggle--large',
      }
   }
  }
}


const Template = args => Toggle(args);

export const Standard = Template.bind({});
Standard.args = {}

export const WithLabel = Template.bind({});
WithLabel.args = {
   label: "Toggle Switch"
}

export const Selected = Template.bind({});
Selected.args = {
   checked: true
}

export const Large = Template.bind({});
Large.args = {
   variation: "toggle toggle--large"
}