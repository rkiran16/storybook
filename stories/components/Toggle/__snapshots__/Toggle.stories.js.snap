// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Toggle Large 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    ">
        <input class=\\"form-check-input\\" ?disabled=",
    " ?checked=",
    " type=\\"checkbox\\" id=\\"flexSwitchCheckDefault\\" aria-label=\\"Toggle Switch\\" aria-hidden=\\"true\\">
        ",
    " 
    </div>
   ",
  ],
  "values": Array [
    "form-check form-switch toggle toggle--large",
    false,
    false,
    "",
  ],
}
`;

exports[`Storyshots Components/Toggle Selected 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    ">
        <input class=\\"form-check-input\\" ?disabled=",
    " ?checked=",
    " type=\\"checkbox\\" id=\\"flexSwitchCheckDefault\\" aria-label=\\"Toggle Switch\\" aria-hidden=\\"true\\">
        ",
    " 
    </div>
   ",
  ],
  "values": Array [
    "form-check form-switch toggle",
    false,
    true,
    "",
  ],
}
`;

exports[`Storyshots Components/Toggle Standard 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    ">
        <input class=\\"form-check-input\\" ?disabled=",
    " ?checked=",
    " type=\\"checkbox\\" id=\\"flexSwitchCheckDefault\\" aria-label=\\"Toggle Switch\\" aria-hidden=\\"true\\">
        ",
    " 
    </div>
   ",
  ],
  "values": Array [
    "form-check form-switch toggle",
    false,
    false,
    "",
  ],
}
`;

exports[`Storyshots Components/Toggle With Label 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    ">
        <input class=\\"form-check-input\\" ?disabled=",
    " ?checked=",
    " type=\\"checkbox\\" id=\\"flexSwitchCheckDefault\\" aria-label=\\"Toggle Switch\\" aria-hidden=\\"true\\">
        ",
    " 
    </div>
   ",
  ],
  "values": Array [
    "form-check form-switch toggle",
    false,
    false,
    Object {
      "_$litType$": 1,
      "strings": Array [
        "<label class=\\"form-check-label\\" for=\\"flexSwitchCheckDefault\\">",
        "</label>",
      ],
      "values": Array [
        "Toggle Switch",
      ],
    },
  ],
}
`;
